gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiKeyServiceImpl.java

```diff
@@ -403,36 +402,60 @@ public ApiKeyEntity updateDaysToExpirationOnLastNotification(String apiKey, Stri
     }
 
     @Override
-    public boolean canCreate(String apiKey, String apiId, String applicationId) {
-        LOGGER.debug("Check if an API Key can be created for api {} and application {}", apiId, applicationId);
+    public boolean canCreate(String apiKeyValue, SubscriptionEntity subscription) {
+        LOGGER.debug("Check if an API Key can be created for subscription {}", subscription.getId());
+
+        // TODO: make environment a parameter
+        ApplicationEntity application = applicationService.findById(GraviteeContext.getCurrentEnvironment(), subscription.getApplication());
+
         try {
             return apiKeyRepository
-                .findByKey(apiKey)
+                .findByKey(apiKeyValue)
                 .stream()
                 .noneMatch(
                     existingKey ->
-                        !existingKey.getApplication().equals(applicationId) ||
-                        (existingKey.getApplication().equals(applicationId) && existingKey.getApi().equals(apiId))
+                        !existingKey.getApplication().equals(application.getId()) ||
+                        (
+                            existingKey.getApplication().equals(application.getId()) &&
+                            existingKey.getSubscriptions().contains(subscription.getId())
+                        )
                 );
         } catch (TechnicalException ex) {
             String message = String.format(
                 "An error occurs while checking if API Key can be created for api %s and application %s",
-                apiId,
-                applicationId
+                subscription.getApi(),
+                subscription.getApplication()
             );
             LOGGER.error(message, ex);
             throw new TechnicalManagementException(message, ex);
         }
     }
 
+    @Override
+    public boolean canCreate(String apiKey, String apiId, String applicationId) {
+        SubscriptionQuery query = new SubscriptionQuery();
+        query.setApi(apiId);
+        query.setApplication(applicationId);
+
+        SubscriptionEntity subscription = subscriptionService
+            .search(query)
+            .stream()
+            .findFirst()
+            .orElseThrow(
+                () -> new TechnicalManagementException("Unable to find subscription for API " + apiId + " and application " + applicationId)
```

Comment:
Here, I think this is normal that we don't find an existing subscription.
Cause we call  `canCreate` before creating a new subscription, to ensure the custom api key is valid for it.

I think this modification causes a regression on the custom API key feature.
We can discuss about it.